File: DotNet/SETPractice/SOLIDPrinciples/LiskovSubstitution/CurrentAccount.cs
namespace SOLIDPrinciples.LiskovSubstitution
{
    public class CurrentAccount : BankAccount
    {
        public decimal OverdraftLimit { get; set; }

        public CurrentAccount(string accountNumber, decimal balance, decimal overdraftLimit) : base(accountNumber, balance)
        {
            OverdraftLimit = overdraftLimit;
        }

        public override void Withdraw(decimal amount)
        {
            if(amount <= Balance + OverdraftLimit)
            {
                Balance -= amount;
                Console.WriteLine($"Remaining Balance in your Current Account is {Balance}");
            }
            else
                Console.WriteLine("Insufficient Balance in your Current Account");

        }
    }
}

File: Java/SETPractice/src/main/java/SOLIDPrinciple/LiskovSubstitution/CurrentAccount.java
package SOLIDPrinciple.LiskovSubstitution;

public class CurrentAccount extends BankAccount{
    public double OverdraftLimit;
    public CurrentAccount(String accountNumber, double balance, double overdraftLimit) {
        super(accountNumber, balance);
        OverdraftLimit = overdraftLimit;
    }

    @Override
    public void Withdraw(double amount) {
        if(amount <= Balance + OverdraftLimit)
        {
            Balance -= amount;
            System.out.println("Remaining Balance in your Current Account is " + Balance);
        }
        else
            System.out.println("Insufficient Balance in your Current Account");
    }
}
